crates/revm/src/evm_impl.rs

```diff
@@ -396,9 +396,13 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
             }
         };
 
+        let bytecode = Bytecode::new_raw(inputs.init_code.clone());
+        let hash = bytecode.hash_slow();
+
         let contract = Box::new(Contract::new(
             Bytes::new(),
-            Bytecode::new_raw(inputs.init_code.clone()),
+            bytecode,
+            hash,
```

Comment:
There is `code_hash` variable above that can be used here. Doing keccak two times is not needed.